format/leveldb/leveldb_table.go

```diff
@@ -360,8 +365,8 @@ func readInternalKey(sharedBytes []byte, unsharedSize int, d *decode.D) error {
 
 	// case 2: type and sequence_number fit fully in unshared: simulate user_key value.
 	if unsharedSize >= typeAndSequenceNumberSize {
-		br := d.FieldRawLen("user_key_suffix", int64(unsharedSize-typeAndSequenceNumberSize)*8)
-		d.FieldValueStr("user_key", string(append(sharedBytes, d.ReadAllBits(br)...)), strInferred)
+		suffix := fieldUTF8ReturnBytes("user_key_suffix", unsharedSize-typeAndSequenceNumberSize, d)
+		d.FieldValueStr("user_key", stringify(sharedBytes, suffix), strInferred)
```

Comment:
Ah, interesting point with UTF8BOM (0xEF, 0xBB, 0xBF)!

I hadn't come across this before nor thought about it.
It seems the Python encoder natively skips it (unless specified explicitly), also some of the LevelDB dumps I've observed so far don't use the UTF8BOM.
That said, since LevelDB takes bytestrings as keys and values, that wouldn't stop anyone from using UTF8BOM if they wanted to.

So maybe the utf-8 encoder will be good enough in most cases?

Reply:
Safest i think is probably to keep it as is, treat the parts as raw bytes. It's a bit weird that UTF8 with a BOM is even a thing as UTF8 does not really have a byte order :) but it seems to happen 🤷 the encoding used by `d.UTF8` is choose to be quite liberal with weird stuff :)

BTW if you want to have full control over text encoding then there is `d.FieldStr("...", <encoding>)` where you can plugin any `encoding.Encoding`.